framework/src/araui/painter/ui_painter.c

```diff
@@ -0,0 +1,602 @@
+					x--;
+					epsilon = epsilon + 8 * (y - x) + 4;
+				} else {
+					epsilon = epsilon + 8 * y + 4;
+				}
+				ui_hal_draw_pixel(center_x + x, center_y + y, color);
+			}
+			epsilon = 4 * (x - 1) * (x - 1) + (2 * y + 1) * (2 * y + 1) - 4 * radius * radius;
+			while (x >= end_x) {
+				x--;
+				if (epsilon < 0) {
+					y++;
+					epsilon = epsilon + 8 * (y - x) + 4;
+				} else {
+					epsilon = epsilon - 8 * x + 4;
+				}
+				ui_hal_draw_pixel(center_x + x, center_y + y, color);
+			}
+			break;
+		case 2:
+			epsilon = 4 * (x - 1) * (x - 1) + (2 * y - 1) * (2 * y - 1) - 4 * radius * radius;
+			while (-1 * x <= y && x > end_x) {
+				x--;
+				if (epsilon > 0) {
+					y--;
+					epsilon = epsilon - 8 * (x + y) + 4;
+				} else {
+					epsilon = epsilon - 8 * x + 4;
+				}
+				ui_hal_draw_pixel(center_x + x, center_y + y, color);
+			}
+			epsilon = (2 * x - 1) * (2 * x - 1) + 4 * (y - 1) * (y - 1) - 4 * radius * radius;
+			while (y >= end_y) {
+				y--;
+				if (epsilon < 0) {
+					x--;
+					epsilon = epsilon - 8 * (x + y) + 4;
+				} else {
+					epsilon = epsilon - 8 * y + 4;
+				}
+				ui_hal_draw_pixel(center_x + x, center_y + y, color);
+			}
+			break;
+		case 3:
+			epsilon = (2 * x + 1) * (2 * x + 1) + 4 * (y - 1) * (y - 1) - 4 * radius * radius;
+			while (y >= x && y > end_y) {
+				y--;
+				if (epsilon > 0) {
+					x++;
+					epsilon = epsilon + 8 * (x - y) + 4;
+				} else {
+					epsilon = epsilon - 8 * y + 4;
+				}
+				ui_hal_draw_pixel(center_x + x, center_y + y, color);
+			}
+			epsilon = 4 * (x + 1) * (x + 1) + (2 * y -1) * (2 * y - 1) - 4 * radius * radius;
+			while (x <= end_x) {
+				x++;
+				if (epsilon < 0) {
+					y--;
+					epsilon = epsilon + 8 * (x - y) + 4;
+				} else {
+					epsilon = epsilon + 8 * x + 4;
+				}
+				ui_hal_draw_pixel(center_x + x, center_y + y, color);
+			}
+			break;
+		case 4:
+			epsilon = 4 * (x + 1) * (x + 1) + (2 * y + 1) * (2 * y + 1) - 4 * radius * radius;
+			while (x <= -1 * y && x < end_x) {
+				x++;
+				if (epsilon > 0) {
+					y++;
+					epsilon = epsilon + 8 * (x + y) + 4;
+				} else {
+					epsilon = epsilon + 8 * x + 4;
+				}
+				ui_hal_draw_pixel(center_x + x, center_y + y, color);
+			}
+			epsilon = (2 * x + 1) * (2 * x + 1) + 4 * (y + 1) * (y + 1) - 4 * radius * radius;
+			while (y <= end_y) {
+				y++;
+				if (epsilon < 0) {
+					x++;
+					epsilon = epsilon + 8 * (x + y) + 4;
+				} else {
+					epsilon = epsilon + 8 * y + 4;
+				}
+				ui_hal_draw_pixel(center_x + x, center_y + y, color);
+			}
+			break;
+		default:
+			break;
+	}
+}
+
+static int _ui_painter_get_quadrant(int x, int y)
+{
+	if (x > 0 && y >= 0) {
+		return 1;
+	}
+
+	if (x <= 0 && y > 0) {
+		return 2;
+	}
+
+	if (x < 0 && y <= 0) {
+		return 3;
+	}
+
+	if (x >= 0 && y < 0) {
+		return 4;
+	}
+
+	//TO DO: Error message will be printed and return error value (INVALID_PARAM)
+	return -1;
+}
+
+static int _ui_painter_get_next_quadrant(int quadrant)
+{
+	switch (quadrant) {
```

Comment:
Circular. you can use (return (value + K) % number_of_item) for this kind of style.

Reply:
DIfferent. 'default' value is -1.